lib/value.ml

```diff
@@ -393,15 +394,21 @@ let of_yaml_record_to_expr ~loc fields =
       ]
   in
   let option_to_none t =
-    match t.pld_type with
-    | [%type: [%t? _] option] -> [%expr Ok None]
-    | _ ->
-        [%expr
-          Error
-            (`Msg
-              [%e
-                estring ~loc
-                  ("Didn't find the function for key: " ^ t.pld_name.txt)])]
+    match Attribute.get Attrs.default t with
+    | None -> (
+        match t.pld_type with
+        | [%type: [%t? _] option] -> [%expr Ok None]
+        | _ ->
+            [%expr
+              Error
+                (`Msg
+                  [%e
+                    estring ~loc
+                      ("Didn't find the function for key: " ^ t.pld_name.txt)])]
+        )
+    | Some default ->
+        let default = [%expr ([%e default] : [%t t.pld_type])] in
```

Comment:
Oh, that looks interesting! I'm curious now: what's the reason why we need this line?

Reply:
I don't think it is necessary now that you point it out :)) I think I thought it might lead to better error messages or something, but I've just tried it out and it isn't necessary.